web/libs/app-common/src/blocks/StorageProviderForm/components/provider-form.tsx

```diff
@@ -23,6 +24,23 @@ export const ProviderForm: React.FC<ProviderFormProps> = ({
 }) => {
   return (
     <div className="space-y-6">
+      {/* Render hidden fields first, outside of layout */}
+      {provider.fields
+        .filter((field) => field.type === "hidden" && (!target || !field.target || field.target === target))
```

Comment:
better to use function with semantic name here for better readability and easy use